src/utils/sentry/httpclient.ts

```diff
@@ -0,0 +1,486 @@
+    },
+  }
+}) as IntegrationFn
+
+/**
+ * Create events for failed client side HTTP requests.
+ */
+export const httpClientIntegration = defineIntegration(_httpClientIntegration)
+
+/**
+ * Interceptor function for fetch requests
+ *
+ * @param requestInfo The Fetch API request info
+ * @param response The Fetch API response
+ * @param requestInit The request init object
+ */
+function _fetchResponseHandler(
+  options: HttpClientOptions,
+  requestInfo: RequestInfo,
+  response: Response,
+  requestInit?: RequestInit,
+): void {
+  const shouldCapture = _shouldCaptureResponse(options, response.status, response.url)
+  if (shouldCapture) {
+    const request = _getRequest(requestInfo, requestInit)
+
+    let requestHeaders, responseHeaders, requestCookies, responseCookies
+
+    if (_shouldSendDefaultPii()) {
+      ;[
+        { headers: requestHeaders, cookies: requestCookies },
+        { headers: responseHeaders, cookies: responseCookies },
+      ] = [
+        { cookieHeader: 'Cookie', obj: request },
+        { cookieHeader: 'Set-Cookie', obj: response },
+      ].map(({ cookieHeader, obj }) => {
+        const headers = _extractFetchHeaders(obj.headers)
+        let cookies
+
+        try {
+          const cookieString =
+            headers[cookieHeader] || headers[cookieHeader.toLowerCase()] || undefined
+
+          if (cookieString) {
+            cookies = _parseCookieString(cookieString)
+          }
+        } catch (e) {
+          logger.log(`Could not extract cookies from header ${cookieHeader}`)
+        }
+
+        return {
+          headers,
+          cookies,
+        }
+      })
+    }
+
+    const event = _createEvent({
+      request: {
+        url: request.url,
+        method: request.method,
+        requestHeaders,
+        requestCookies,
+        data: requestInit?.body,
+        query_string: request.url.includes('?') ? request.url.split('?')[1] : undefined,
+      },
+      response: {
+        status: response.status,
+        responseHeaders,
+        responseCookies,
+      },
+    })
+
+    captureEvent(event)
+  }
+}
+
+/**
+ * Interceptor function for XHR requests
+ *
+ * @param xhr The XHR request
+ * @param method The HTTP method
+ * @param headers The HTTP headers
+ */
+function _xhrResponseHandler(
+  options: HttpClientOptions,
+  xhr: SentryWrappedXMLHttpRequest & XMLHttpRequest,
+  method: string,
+  headers: Record<string, string>,
+): void {
+  const shouldCapture = _shouldCaptureResponse(options, xhr.status, xhr.responseURL)
+
+  if (shouldCapture) {
+    let requestHeaders, responseCookies, responseHeaders
+
+    if (_shouldSendDefaultPii()) {
+      try {
+        const cookieString =
+          xhr.getResponseHeader('Set-Cookie') || xhr.getResponseHeader('set-cookie') || undefined
+
+        if (cookieString) {
+          responseCookies = _parseCookieString(cookieString)
+        }
+      } catch (e) {
+        logger.log('Could not extract cookies from response headers')
```

Comment:
do we want to include the original error message when logging?

Reply:
I'd probably not touch this bit as this is from the base implementation - not sure how much we can get out of that error, and didn't manage to even get into it so it may just be paranoia from sentry's implementation end